sbnana/CAFAna/Systs/EnergySysts.h

```diff
@@ -0,0 +1,240 @@
+//ETW May 2020 Fresh version for SBN
+
+#pragma once
+
+#include "sbnana/CAFAna/Core/ISyst.h"
+#include "sbnana/CAFAna/Core/Utilities.h"
+#include "sbnana/CAFAna/Analysis/ExpInfo.h"
+#include "sbnanaobj/StandardRecord/Proxy/SRProxy.h"
+
+#include "TFile.h"
+#include "TH1.h"
+#include "TRandom3.h"
+
+#include <cassert>
+
+namespace ana
+{
+
+  enum class EnergyScaleSystTerm {
+    Constant,
+    Sqrt,
+    InverseSqrt
+  };
+
+  enum class EnergyScaleSystParticle {
+    Muon,
+    Hadron,
+    Neutron,
+    EM,
+    ChargedHadron
+  };
+
+  enum class EnergyScaleSystDetector {
+    SBND,
+    MicroBooNE,
+    ICARUS,
+    All
+  };
+
+  template<EnergyScaleSystTerm term, EnergyScaleSystParticle part, EnergyScaleSystDetector detector = EnergyScaleSystDetector::All>
```

Comment:
Does this templating really gain you anything over just having parameters to the constuctor and members like `fTerm`?

Reply:
Makes sense. I think variables are more familiar, and templates always end up causing pain that means they're not worth it unless you need extreme generality.